processing/src/main/java/io/druid/query/extraction/namespace/URIExtractionNamespace.java

```diff
@@ -0,0 +1,508 @@
+    return toString().equals(namespace1.toString());
+  }
+
+  @Override
+  public int hashCode()
+  {
+    int result = namespace.hashCode();
+    result = 31 * result + uri.hashCode();
+    result = 31 * result + namespaceParseSpec.hashCode();
+    result = 31 * result + pollPeriod.hashCode();
+    result = 31 * result + (versionRegex != null ? versionRegex.hashCode() : 0);
+    return result;
+  }
+
+
+  private static class DelegateParser implements Parser<String, String>
+  {
+    private final Parser<String, Object> delegate;
+    private final String key;
+    private final String value;
+
+    private DelegateParser(
+        Parser<String, Object> delegate,
+        String key,
+        String value
+    )
+    {
+      this.delegate = delegate;
+      this.key = key;
+      this.value = value;
+    }
+
+    @Override
+    public Map<String, String> parse(String input)
+    {
+      final Map<String, Object> inner = delegate.parse(input);
+      final String k = (String) Preconditions.checkNotNull(
+          inner.get(key),
+          "Key column [%s] missing data in line [%s]",
+          key,
+          input
+      );
+      final String val = (String) Preconditions.checkNotNull(
+          inner.get(value),
+          "Value column [%s] missing data in line [%s]",
+          value,
+          input
+      );
+      return ImmutableMap.<String, String>of(k, val);
+    }
+
+    @Override
+    public void setFieldNames(Iterable<String> fieldNames)
+    {
+      delegate.setFieldNames(fieldNames);
+    }
+
+    @Override
+    public List<String> getFieldNames()
+    {
+      return delegate.getFieldNames();
+    }
+  }
+
+  @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+  @JsonSubTypes(value = {
+      @JsonSubTypes.Type(name = "csv", value = CSVFlatDataParser.class),
+      @JsonSubTypes.Type(name = "tsv", value = TSVFlatDataParser.class),
+      @JsonSubTypes.Type(name = "customJson", value = JSONFlatDataParser.class),
+      @JsonSubTypes.Type(name = "simpleJson", value = ObjectMapperFlatDataParser.class)
+  })
+  public static interface FlatDataParser
+  {
+    Parser<String, String> getParser();
+  }
+
+  @JsonTypeName("csv")
+  public static class CSVFlatDataParser implements FlatDataParser
+  {
+    private final Parser<String, String> parser;
+    private final List<String> columns;
+    private final String keyColumn;
+    private final String valueColumn;
+
+    @JsonCreator
+    public CSVFlatDataParser(
+        @JsonProperty("columns") List<String> columns,
+        @JsonProperty("keyColumn") final String keyColumn,
+        @JsonProperty("valueColumn") final String valueColumn
+    )
+    {
+      Preconditions.checkArgument(
+          Preconditions.checkNotNull(columns, "`columns` list required").size() > 1,
+          "Must specify more than one column to have a key value pair"
+      );
+      this.parser = new DelegateParser(new CSVParser(Optional.<String>absent(), columns), keyColumn, valueColumn);
```

Comment:
but this will propagate null values of `keyColumn` and `valueColum` if the user is using the default configuration. this line of code should be after the line 270, also we have to pass `this.keyColumn` and `this.valueColum`